src/github/tests/test_utils.rs

```diff
@@ -176,6 +250,94 @@ impl TeamDataBuilder {
     }
 }
 
+#[derive(Clone, Builder)]
+#[builder(pattern = "owned")]
+pub struct RepoData {
+    name: String,
+    #[builder(default)]
+    pub description: String,
+    #[builder(default)]
+    pub homepage: Option<String>,
+    #[builder(default)]
+    bots: Vec<Bot>,
+    #[builder(default)]
+    pub teams: Vec<v1::RepoTeam>,
+    #[builder(default)]
+    pub members: Vec<v1::RepoMember>,
+    #[builder(default)]
+    pub archived: bool,
+    #[builder(default)]
+    pub allow_auto_merge: bool,
+}
+
+impl RepoData {
+    pub fn new(name: &str) -> RepoDataBuilder {
+        RepoDataBuilder::default().name(name.to_string())
+    }
+
+    pub fn add_member(&mut self, name: &str, permission: RepoPermission) {
+        self.members.push(v1::RepoMember {
+            name: name.to_string(),
+            permission,
+        });
+    }
+
+    pub fn add_team(&mut self, name: &str, permission: RepoPermission) {
+        self.teams.push(v1::RepoTeam {
+            name: name.to_string(),
+            permission,
+        });
+    }
+
+    fn to_data(&self) -> v1::Repo {
+        let RepoData {
+            name,
+            description,
+            homepage,
+            bots,
+            teams,
+            members,
+            archived,
+            allow_auto_merge,
+        } = self.clone();
+        v1::Repo {
+            org: DEFAULT_ORG.to_string(),
+            name: name.clone(),
+            description,
+            homepage,
+            bots,
+            teams: teams.clone(),
+            members: members.clone(),
+            branch_protections: vec![],
+            archived,
+            private: false,
+            auto_merge_enabled: allow_auto_merge,
+        }
+    }
+}
```

Comment:
I wonder why you didn't use the From trait

Reply:
Good idea, we'll need to create the `v1::Repo` for branch protections tests later anyway.